src/Util/HTTPSignature.php

```diff
@@ -604,13 +604,14 @@ public static function getKeyIdContact(array $http_headers): array
 	/**
 	 * Gets a signer from a given HTTP request
 	 *
-	 * @param string $content
-	 * @param array $http_headers
+	 * @param string   $content
+	 * @param array    $http_headers
+	 * @param ?boolean $update true = always update, false = never update, null = update when not found or outdated
 	 *
 	 * @return string|null|false Signer
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 */
-	public static function getSigner(string $content, array $http_headers)
+	public static function getSigner(string $content, array $http_headers, bool $update = null)
```

Comment:
Nullable types without `?` will start throwing deprecation warnings in PHP 8.4.

```suggestion
	public static function getSigner(string $content, array $http_headers, ?bool $update = null)
```

Reply:
Changed